app/models/reminder.rb

```diff
@@ -0,0 +1,47 @@
+class Reminder < ApplicationRecord
+  belongs_to :assignment
+  belongs_to :volunteer
+
+  default_scope { order(created_at: :desc) }
+
+  def self.create_for(volunteer, assignment)
+    Reminder.create! do |r|
+      r.volunteer = volunteer
+      r.assignment = assignment
+    end
+  end
+
+  def self.conditionally_create_reminders(time = Time.zone.now)
+    logger.info("Beginning reminder run, reference Date #{time}")
+    logger.flush
+
+    reminders_created_count = 0
+    Volunteer.all.find_each.count do |volunteer|
+      reminders_created_count = volunteer.assignments.each do |assignment|
+        conditionally_create_reminder_for_volunteer(volunteer, assignment)
+      end
+    end
+
+    logger.info("Created #{reminders_created_count} reminders, reference Date #{time}")
+    logger.flush
+
+    reminders_created_count
+  end
+
+  def self.conditionally_create_reminder_for_volunteer(volunteer, assignment)
+    log_message = if assignment.period_start > Time.zone.today - 6.months
```

Comment:
I prefer this style to keep the indentation consistent, but it's definitely a matter of taste:

```ruby
log_message =
  if ...
    ...
  elsif
    ...
  end
```

Reply:
Also you can just do `6.months.ago`